src/main/java/bio/terra/service/dataset/DatasetSummary.java

```diff
@@ -90,18 +96,56 @@ public DatasetSummary storage(List<? extends StorageResource<?, ?>> storage) {
     return this;
   }
 
+  public List<BillingProfileModel> getBillingProfiles() {
+    return billingProfiles;
+  }
+
+  public DatasetSummary billingProfiles(List<BillingProfileModel> billingProfiles) {
+    this.billingProfiles = billingProfiles;
+    return this;
+  }
+
+  public BillingProfileModel getDefaultBillingProfile() {
+    return billingProfiles.stream()
+        .filter(b -> b.getId().equals(defaultProfileId))
+        .findFirst()
+        .orElseThrow();
+  }
+
   public CloudRegion getStorageResourceRegion(CloudResource storageResource) {
+    return getCloudResourceAttribute(storageResource, StorageResource::getRegion);
+  }
+
+  public boolean datasetStorageContainsRegion(GoogleRegion region) {
+    return storage.stream().anyMatch(sr -> sr.getRegion().equals(region));
+  }
+
+  public CloudPlatform getStorageCloudPlatform() {
+    // A Dataset should not have both a bucket and a storage account at this point
+    return storage.stream()
```

Comment:
Again, technically it can for now

Reply:
How do we reconcile where we're trying to ingest a file if we have both a Google and Azure storage resource?

Reply:
I think it's a weird in between state as we build this out.  I've been thinking on this and I think it's safe to make the assumption you're making here